#! /bin/sh -e
## 02_various_fixes.dpatch by Martin Loschwitz <madkiss@madkiss.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes various smaller mistakes

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

--- qt-x11-free-3.2.3.old/tools/designer/integration/designer.el	2003-10-03 13:32:02.000000000 +0200
+++ qt-x11-free-3.2.3/tools/designer/integration/designer.el	2003-12-19 13:28:14.000000000 +0100
@@ -1,9 +1,9 @@
-(defvar designer-exec (concat (getenv "QTDIR") "/bin/designer"))
+(defvar designer-exec "/usr/bin/designer")
 
 (defun designer-open-ui-file ()
   (interactive)
   (defvar procname "designer")
-  (start-process procname nil designer-exec "-style" "windows" "-client" buffer-file-name)
+  (start-process procname nil designer-exec "-client" buffer-file-name)
 )
 
 (defun designer-create-ui-file (&optional arg)
diff -ruN qt-x11-free-3.2.3.old/tools/tools.pro qt-x11-free-3.2.3/tools/tools.pro
--- qt-x11-free-3.2.3.old/tools/tools.pro	2003-09-24 13:32:33.000000000 +0200
+++ qt-x11-free-3.2.3/tools/tools.pro	2003-12-19 13:26:58.000000000 +0100
@@ -6,7 +6,12 @@
 		      designer \
 		      assistant \
 		      linguist
-    unix:SUBDIRS	+= qtconfig
+    unix:SUBDIRS	+= qtconfig \
+    			mergetr \
+			qconfig \
+			makeqpf \
+			msg2qm \
+			qembed
 }
 
 CONFIG+=ordered
diff -ruN qt-x11-free-3.3.2.old/tools/designer/designer.pro qt-x11-free-3.3.2/tools/designer/designer.pro
--- qt-x11-free-3.3.2.old/tools/designer/designer.pro	2003-10-31 15:36:28.000000000 +0100
+++ qt-x11-free-3.3.2/tools/designer/designer.pro	2004-05-14 16:08:10.000000000 +0200
@@ -5,7 +5,9 @@
 SUBDIRS	=  uic \
 	   uilib \
 	   designer \
-	   app
+	   app \
+	   tools/createcw \
+	   tools/conv2ui
 
 dll:SUBDIRS *=  editor plugins
 shared:SUBDIRS *=  editor plugins
diff -ruN qt-x11-free-3.3.3.old/tools/designer/tools/createcw/createcw.pro qt-x11-free-3.3.3/tools/designer/tools/createcw/createcw.pro
--- qt-x11-free-3.3.3.old/tools/designer/tools/createcw/createcw.pro	2003-10-31 14:36:28.000000000 +0000
+++ qt-x11-free-3.3.3/tools/designer/tools/createcw/createcw.pro	2004-08-13 23:19:54.000000000 +0000
@@ -3,3 +3,6 @@
 HEADERS		=
 SOURCES		= main.cpp
 TARGET		= createcw
+
+target.path = $$bins.path
+INSTALLS += target
diff -ruN qt-x11-free-3.3.3.old/tools/makeqpf/makeqpf.pro qt-x11-free-3.3.3/tools/makeqpf/makeqpf.pro
--- qt-x11-free-3.3.3.old/tools/makeqpf/makeqpf.pro	2004-06-08 08:07:19.000000000 +0000
+++ qt-x11-free-3.3.3/tools/makeqpf/makeqpf.pro	2004-08-13 23:20:33.000000000 +0000
@@ -5,3 +5,6 @@
 INTERFACES	=
 TARGET		= makeqpf
 REQUIRES=full-config nocrosscompiler
+
+target.path = $$bins.path
+INSTALLS += target
diff -ruN qt-x11-free-3.3.3.old/tools/mergetr/mergetr.pro qt-x11-free-3.3.3/tools/mergetr/mergetr.pro
--- qt-x11-free-3.3.3.old/tools/mergetr/mergetr.pro	2004-06-08 08:07:26.000000000 +0000
+++ qt-x11-free-3.3.3/tools/mergetr/mergetr.pro	2004-08-13 23:21:09.000000000 +0000
@@ -4,3 +4,6 @@
 SOURCES		= mergetr.cpp
 TARGET		= mergetr
 REQUIRES=full-config nocrosscompiler
+
+target.path = $$bins.path
+INSTALLS += target
diff -ruN qt-x11-free-3.3.3.old/tools/msg2qm/msg2qm.pro qt-x11-free-3.3.3/tools/msg2qm/msg2qm.pro
--- qt-x11-free-3.3.3.old/tools/msg2qm/msg2qm.pro	2004-06-08 08:08:25.000000000 +0000
+++ qt-x11-free-3.3.3/tools/msg2qm/msg2qm.pro	2004-08-13 23:22:13.000000000 +0000
@@ -5,3 +5,6 @@
 SOURCES		= msg2qm.cpp
 TARGET		= msg2qm
 REQUIRES=full-config nocrosscompiler
+
+target.path = $$bins.path
+INSTALLS += target
diff -ruN qt-x11-free-3.3.3.old/tools/qembed/qembed.pro qt-x11-free-3.3.3/tools/qembed/qembed.pro
--- qt-x11-free-3.3.3.old/tools/qembed/qembed.pro	2004-06-08 08:08:40.000000000 +0000
+++ qt-x11-free-3.3.3/tools/qembed/qembed.pro	2004-08-13 23:22:44.000000000 +0000
@@ -2,3 +2,6 @@
 SOURCES = qembed.cpp
 TARGET  = qembed
 REQUIRES=full-config nocrosscompiler
+
+target.path = $$bins.path
+INSTALLS += target
